src/data.py

```diff
@@ -182,23 +182,12 @@ def add_items(self, items: misc.ItemList) -> None:
         style_manager = self.get_application().get_figure_style_manager()
         selected_style = style_manager.get_selected_style_params()
         color_cycle = selected_style["axes.prop_cycle"].by_key()["color"]
-        used_colors = []
-
-        def _append_used_color(color):
-            used_colors.append(color)
-            # If we've got all colors once, remove those from used_colors so we
-            # can loop around
```

Comment:
Removing this breaks the current behaviour that it loops around. So if there's three colours in the settings: Red, Green, Blue. Then in the current main branch it goes like this when adding six items: Red, Green, Blue, Red, Green, Blue
But by removing this logic, it now goes like this: Red, Green, Blue, Red, Red, Red. 

That's a regression imo. Especially if you're adding a lot of data at once, this can become a bit weird with one color suddenly dominating.

Reply:
This is replaced by the logic on line 236, which is responsible for detecting that

Reply:
Nevermind, found the cause of the issue you were describing. Also fixed an issue, that was in main in regards to custom colors (when having a custom color present, the looping wouldn't work either)